### PR TITLE
Corrections in rule chronyd_specify_remote_server

### PR DESCRIPTION
**Pull request by rumch-se**
#### Description:

- _Correction is rule chronyd_specify_remote_server as a part of CIS profile for SLE 12/15 _

#### Rationale:

In the current version:
- The bash part did not add server + list of time servers to /etc/chrony.conf. 
- The variable var_multiple_time_servers was not included in different profiles which belong to SLE 12/15

Note: Please send this PR to Vojtech Polasek for review. Probably "grep -q" returns empty string and because of that every execution of remediation will add server to /etc/chrony.conf. Also I am not sure if the remediation will add the word server in from of list of time servers in  /etc/chrony.conf.

### CODE REVIEW

**Comment by rumch-se** [comment]
Hello @vojtapolasek,
May you check my note for this PR.
Have a nice day
Rumen

File: linux_os/guide/services/ntp/chronyd_specify_remote_server/bash/shared.sh [hunk 1]
```diff
@@ -4,6 +4,15 @@
 
 config_file="{{{ chrony_conf_path }}}"
 
+{{% if product in ["sle12", "sle15"] %}}
```
Comment (vojtapolasek) [inline linux_os/guide/services/ntp/chronyd_specify_remote_server/bash/shared.sh:7]: @rumch-se  what is actually wrong with the remediation which is already in place? I tested it and it works well, it adds server entries. The remediation you added does not work unfortunately - no entries were added.

**Comment by rumch-se** [comment]
Hello @vojtapolasek 
1/
The definition in CIS of this rule is the following:
+++
Audit 
# grep -E "^(server|pool)" /etc/chrony.conf
server <remote-server>

Remediation:
Add or edit server or pool lines to /etc/chrony.conf as appropriate:
server <remote-server>
+++
In out bug system at SUSE we have a bug related to this rule, and the details which I have are.   
++
Currently implemented regex in remediation does not work, servers re-added on every run
++
Note: The bug was registered before my change i.e. this bug should correspond to the rule before my PR.
2/
With the version for SUSE OS I am checking for existence of a row in /etc/chrony.conf and I add it if the row does not exist 
If  I run remediation - 2+ times and row is already in crony.conf - it should not be added.
The row is in /etc/chrony.conf  should be 
server NAMES_OF_SERVERS 
where the NAMES_OF_SERVERS should be taken from variable
3/
Did you test to run remediation few times, not only once ?
Did you check/review /etc/chrony.conf  after each remediation?  
When I executed in bash var=$(grep -q '^[\s]*(?:server|pool)[\s]+[\w]+' "$config_file") or 
var=$(grep -q '^[\s]*(?:server|pool)[\s]+[\w]+' /etc/chrony.conf) and after echo $var - I did not have anything on my screen.

in /etc/chrony.conf I have at the bottom:
....
# Also include any directives found in configuration files in /etc/chrony.d
include /etc/chrony.d/*.conf
server 0.suse.pool.ntp.org,1.suse.pool.ntp.org,2.suse.pool.ntp.org,3.suse.pool.ntp.org

.. and the result is 
sles-15-sp2:~ # var=$(grep -q '^[\s]*(?:server|pool)[\s]+[\w]+' /etc/chrony.conf)
sles-15-sp2:~ # echo $var

sles-15-sp2:~ # 
++ 
BUT with my approach 
sles-15-sp2:~ # test=$(grep -E "^(server|pool)" /etc/chrony.conf)
sles-15-sp2:~ # echo $test
server 0.suse.pool.ntp.org,1.suse.pool.ntp.org,2.suse.pool.ntp.org,3.suse.pool.ntp.org
sles-15-sp2:~ #

**Comment by vojtapolasek** [comment]
@rumch-se  now I see the problem. The original remediation is indeed wrong - it will add new server entries no matter if there are already defined some entries before.
I suggest you merge your suggestion into existing one so that you fix it for everybody. At the same time, your approach creates invalid server directive. Each server directive can have only one server defined, at least that's what I understand from the man page. But you can get inspired by the existing for loop.

**Comment by rumch-se** [comment]
Hello @vojtapolasek 
I have corrected bash and ansible parts of the rule.
Now after execution of the remediation you will have rows in /etc/chrony.conf like this 
server 0.suse.pool.ntp.org
server 1.suse.pool.ntp.org
server 2.suse.pool.ntp.org
server 3.suse.pool.ntp.org
Please check my solution in bash  - because I have a loop based on servers defined in the variable and I do a check if such a string (server) exists (if no - a row will be added) ( if yes - a row will be added if the line does not start with the word server)
I tried to apply this approach in ansible, but I did not find a way to make a check inside the loop, and because of that I replicated the solution from bash into ansible via artificially created .sh file used only for remediation.
If you have better workable idea for ansible - I will appreciate it.
I did not checked OVAL (audit), I hope that it works. 

Have a nice day
Rumen

File: linux_os/guide/services/ntp/chronyd_specify_remote_server/bash/shared.sh [hunk 2]
```diff
@@ -4,6 +4,13 @@
 
 config_file="{{{ chrony_conf_path }}}"
 
-if ! grep -q '^[\s]*(?:server|pool)[\s]+[\w]+' "$config_file" ; then
```
Comment (vojtapolasek) [inline linux_os/guide/services/ntp/chronyd_specify_remote_server/bash/shared.sh:7]: Hello @rumch-se , I have analyzed the expression and I think that the only thing which needs to be fixed is this line (regarding Bash).
There should be:
```
if ! grep -q '^[[:space:]]*\(server\|pool\)[[:space:]]\+[[:graph:]]\+' "$config_file" ; then
```
The grep expression will then behave as we would expect.

**Comment by vojtapolasek** [comment]
With regards to your modification of Ansible... are you sure it needs fixing? I tried to run it multiple times and it puts servers in /etc/chrony.conf only once. That's expected. Did you find any other bugs in the Ansible remediation?

**Comment by vojtapolasek** [comment]
Hello @rumch-se  I don't think that the rule should check if /etc/chrony.conf contains exactly servers which are written in the XCCDF variable. This is not mandated by the CIS profile. The profile does not name exact servers at all. Moreover, some users might not want to use those servers for whatever reasons. The important thing is that there IS some server configured.
Therefore, I think that the only part of this rule which needs fixing is that regex in my previous comment.
Do you have any profile which would mandate presence of particular servers,

**Comment by rumch-se** [comment]
Hello @vojtapolasek

Configuration of time servers is very very important in DoD (Department of Deference) ex:  RedHat STIG ID: RHEL-08-030740 , SUSE SLES-15-010400, Banking industry and other strongly regulated industries (Insurance, Healthcare)  . In general in Enterprise environment one of the main controls which all auditors is looking is this (in my professional life I passed 10+ audits from 4Bigs and Central Bank of my country)  The main reason - in case when your environment is compromised - you have to be able to provide a digital evidence and time synchronization is very important for digital forensics. Otherwise it will be very difficult to investigate the relationship between different logs. Because of that - usually we have a central log location (protected from the administrators) and all machines should have time synchronization from a least two independent sources of time.  Usually there is an additional security control, subject of regular audit, about your time sources.   With this rule we are trying to fix / remediate a system and to put it in state that it will have a correct time servers. Also all Intrusion detection / prevention systems which make correlation by using logs from the central log collector will rely on this information. 

For sure we can put what we want in the variable, but imagine that the remediation has to be applied in a data center with 1000+ hosts. This will impact negatively the whole infrastructure. Because of that the rule should apply as much as possible the definitions of the time servers located into the variable. 

Have a nice day
Rumen

**Comment by vojtapolasek** [comment]
Hello @rumch-se , I am not sure if there is not a misunderstanding here. I will try to rephrase my concerns.
The current state of the rule chronyd_specify_remote_server is like this:
- the OVAL checks only for the pool or server directive and for the fact that it is followed by one or more spaces and then one or more character represented by \w. It does NOT consulte the var_multiple_time_servers XCCDF variable at all.
- Ansible remediation first does the same check as OVAL does and if the server or pool directive is not found, it will insert multiple entries for servers presented in the var_multiple_time_servers XCCDF variable
- the Bash remediation should work in the same way as Ansible remediation but it is currently broken, you discovered this correctly and I definitely agree this should be fixed.

I believe that this rule is overall fine and it should stay like this.

From your comment above, I read that you would like to have the rule in this state:
- the OVAL checks for the presence of the server or pool directive followed by one or more spaces and followed by one of servers defined in the var_multiple_time_servers XCCDF variable.
- Ansible remediation first does the same check as OVAL does and it modifies the file in a way so that it contains only servers from the var_multiple_time_servers variable
- the Bash remediation copies behavior of the Ansible directive.

Could you confirm that I understand your desired changes correctly?
Note: In general, I believe the rule  chronyd_specify_remote_server should be just fixed but othervise kept in tact. If you wish for some more stricter rule alonng the lines above, I think this should be introduced as a new rule.

**Comment by rumch-se** [comment]
Good morning @vojtapolasek

Your understood my idea. 
- I don't know if will be easy OVAL to check for one of the servers which is included in the var_multiple_time_servers variable. If it can - it will be the best. At least OVAL can check for a server or pool word followed by a space and after it  for regexp template/pattern which corresponds to a server definitions into var_multiple_time_servers variable. The pattern could be defined as a intersection of all time servers for all vendors. Like that we will be sure that we have a record in .conf file which is correct from a format point of view.
- Ansible  remediation first does the same check as OVAL does and it modifies the file in a way so that it contains only servers from the var_multiple_time_servers variable
- the Bash remediation copies behavior of the Ansible directive.               

Have  a nice day
Rumen

**Comment by rumch-se** [comment]
Hello @vojtapolasek 
I have worked on regexp pattern for time servers, and I think we have something like this 

([0-3]\b(.fedora|.suse|.rhel)\b\b(.pool.ntp.org)\b)|([0-3]\b(.pool.ntp.org)\b)|([0-3]\b(.ntp.cloud.aliyuncs.com)\b)|([0-3]\b(.us.pool.ntp.mil)\b)

1st matching group 
- ([0-3]\b(.fedora|.suse|.rhel)\b\b(.pool.ntp.org)\b) for servers like 0.fedora.pool.ntp.org, 0.rhel.pool.ntp.org and 0.suse.pool.ntp.org
2nd matching group
- ([0-3]\b(.pool.ntp.org)\b) - for servers like 0.pool.ntp.org
3rd matching group 
- ([0-3]\b(.ntp.cloud.aliyuncs.com)\b) - for servers like 0.ntp.cloud.aliyuncs.com
4rt matching group  
- ([0-3]\b(.us.pool.ntp.mil)\b) for servers like 0.us.pool.ntp.mil

Have a nice day
Rumen

**Comment by vojtapolasek** [comment]
@rumch-se  hello, I am glad we are on the same page now. I have two points:
1. If you want to change the OVAL check in such a dramatic way, I suggest you do not touch the original rule and create a new one. Although this stricter check might be useful for some profile (but see my concerns below), I think it should not be forced upon more benevolent profiles such as CIS.
2. Although I think I understand your goal, I think that you can't achieve it with regex. How would you handle the case if a customer would like to use a server which is not mentioned in the variable at all (e.g. their internal server)? I think that the way to go would be to write oval in a way so that it checks presence of exact server names which are in the XCCDF variable. The user would be able to change the variable manually by tailoring. This is allowed because if you look into the file var_multiple_time_servers.var, you will see
```
interactive: true
```
that means that the variable can be manually tailored by user to any value.
@marcusburghardt  @yuumasato  @Mab879  what do you think?

**Comment by rumch-se** [comment]
Hello @vojtapolasek 
We can "remain door open". According to https://chrony.tuxfamily.org/doc/3.4/chrony.conf.html we 
have 
++
server hostname [option]…​

    The server directive specifies an NTP server which can be used as a time source. The client-server relationship is strictly hierarchical: a client might synchronise its system time to that of the server, but the server’s system time will never be influenced by that of a client.

    The server directive is immediately followed by either the name of the server, or its IP address. The server directive supports the following options:
++
in other words we can extend the pattern for regexp with 5 matching group - an IP address

for example this from here https://ihateregex.io/expr/ip/ we have:

(\b25[0-5]|\b2[0-4][0-9]|\b[01]?[0-9][0-9]?)(\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}

In other if customer is planning to use a server which does not corresponds to servers in the variable he will have 3 options:
- to use IP address - this will require changes in bash / ansible to support IP address.
- to use default setting - but in this case he has to play with its internal DNS system and to map internal IP addresses of its time servers to be mapped to servers in default variable. This is possible because usually is a good practice all sensitive machines to be in isolated environment (a subnet) and not to have a direct access to the Internet.   
- to use local hosts file where to define an internal IP addresses which to be mapped to a server(s) names in default variable

Have a nice day
Rumen

**Comment by vojtapolasek** [comment]
Hello @rumch-se  I think this approach is not the correct one. Basically, I think you have two options how to approach this:
1. do not check for the server name at all, just check that something is there (\w+) as used in the original rule
2. check for exact content of XCCDF variables.
Although solutions you offered are interesting, all have problems which I believe we do not want our users to deal with:
quoting:
In other words, if the customer is planning to use a server which does not corresponds to servers in the variable he will have 3 options:

    - to use IP address - this will require changes in bash / ansible to support IP address.

This sounds reasonable, however, you are limiting the user's choice - why should we force them not to use host names? We should actually help them to automate this checks, not force them to change their processes.

    - to use default setting - but in this case he has to play with its internal DNS system and to map internal IP addresses of its time servers to the default servers in var_multiple_time_servers XCCDF variable. This is possible because usually is a good practice all sensitive machines to be in an isolated environment (a subnet) and not to have a direct access to the Internet.
    - to use local hosts file where to define internal IP addresses to be mapped to default server(s) names in var_multiple_time_servers XCCDF variable

This requires the user to perform some hacking on their side.

The more I am thinking about it the more I am leaning towards the idea that we should keep it as it is.
Because although I understand your idea, that you want to prevent the case when the rule passes with any arbitrary string used as a server name / address, the problem is that we can not know if the actual name or address is indeed the valid server in their environment.
I still think that if you want to ensure that there is some predefined string, you should create a new rule which will consider ONLY contents of the XCCDF variable. And then sufficient documentation should be in place for users to know that if they plan to use some custom server names, they should modify the variable.
Actually this rule is a good example:
https://github.com/ComplianceAsCode/content/tree/master/linux_os/guide/system/logging/rsyslog_sending_messages/rsyslog_remote_loghost

**Comment by rumch-se** [comment]
Good morning @vojtapolasek, 

I attached the description of the rule from CIS benchmark for SUSE. Also I attached the rule which I found for Ubuntu 22.04. In the first case the rule is automated, while in the second case is manual. In both cases during the audit we have to verify the server - for example in SUSE we have - verify remote server is configured properly and for Ubuntu we have - Verify that at least one pool line and/or at least three server lines are returned, and the timeserver on the returned lines follows local site policy. According these descriptions we should not check for any arbitrary string. 

Have a nice day
Rumen

[CIS_2_2_1_3.pdf](https://github.com/ComplianceAsCode/content/files/10939655/CIS_2_2_1_3.pdf)
[CIS_Ubuntu_Linux_22_04_chrony.pdf](https://github.com/ComplianceAsCode/content/files/10939657/CIS_Ubuntu_Linux_22_04_chrony.pdf)

**Comment by vojtapolasek** [comment]
Hello @rumch-se , then I believe you should go for the approach mentioned in point 2 of my previous comment. e.g., create a new rule which will check for exact content of the XCCDF variable. I think any other approach will just either not be perfect or not bring good user experience.Hello @rumch-se  I think this approach is not the correct one. Basically, I think you have two options how to approach this:
1. do not check for the server name at all, just check that something is there (\w+) as used in the original rule
2. check for exact content of XCCDF variables.
Although solutions you offered are interesting, all have problems which I believe we do not want our users to deal with:
quoting:
In other words, if the customer is planning to use a server which does not corresponds to servers in the variable he will have 3 options:

    - to use IP address - this will require changes in bash / ansible to support IP address.

This sounds reasonable, however, you are limiting the user's choice - why should we force them not to use host names? We should actually help them to automate this checks, not force them to change their processes.

    - to use default setting - but in this case he has to play with its internal DNS system and to map internal IP addresses of its time servers to the default servers in var_multiple_time_servers XCCDF variable. This is possible because usually is a good practice all sensitive machines to be in an isolated environment (a subnet) and not to have a direct access to the Internet.
    - to use local hosts file where to define internal IP addresses to be mapped to default server(s) names in var_multiple_time_servers XCCDF variable

This requires the user to perform some hacking on their side.

The more I am thinking about it the more I am leaning towards the idea that we should keep it as it is.
Because although I understand your idea, that you want to prevent the case when the rule passes with any arbitrary string used as a server name / address, the problem is that we can not know if the actual name or address is indeed the valid server in their environment.
I still think that if you want to ensure that there is some predefined string, you should create a new rule which will consider ONLY contents of the XCCDF variable. And then sufficient documentation should be in place for users to know that if they plan to use some custom server names, they should modify the variable.
Actually this rule is a good example:
https://github.com/ComplianceAsCode/content/tree/master/linux_os/guide/system/logging/rsyslog_sending_messages/rsyslog_remote_loghost

File: linux_os/guide/services/ntp/chronyd_specify_remote_server/ansible/shared.yml [hunk 3]
```diff
@@ -5,18 +5,41 @@
...
-- name: "Configure remote time servers"
-  lineinfile:
-    path: {{{ chrony_conf_path }}}
-    line: 'server {{ item }}'
-    state: present
-    create: True
-  loop: '{{ var_multiple_time_servers.split(",") }}'
-  when: chrony_servers.matched == 0
+- name: {{{ rule_title }}} Generation of script for remediation
+  ansible.builtin.copy:
```
Comment (marcusburghardt) [inline linux_os/guide/services/ntp/chronyd_specify_remote_server/ansible/shared.yml:14]: I am not fan of this approach of creating shell scripts in Ansible to be executed by the Playbook itself. Ansible tasks should be accomplished by using specific Ansible modules.

**Comment by marcusburghardt** [comment]
I understand the concern from @rumch-se .
It would be good to validade the value of `servers|pool` parameter, but if we want to do this it should be done precisely.
To accomplish this, I don't see any safe way besides checking a XCCDF variable.

In general the Benchmarks are not explicitly saying which servers must be used. They usually mention that it should be properly configured but are not restrict about the values. For these cases, I believe this rule is enough and its logic should not be changed. This PR should be limited to fix the Bash regex, as mentioned in the initial comments.

Therefore, I agree with @vojtapolasek that a new rule should be introduced in a different PR with the capability to consider the value in the XCCDF variable, if we want to precisely check the parameter value. In that case, if an Admin wants to precisely check the value of `server|pool` parameter, he can create a tailored file using this new rule and set the desired value in the variable.

File: linux_os/guide/services/ntp/chronyd_specify_remote_server/ansible/shared.yml (hunk 3 above)
Comment (rumch-se) [reply linux_os/guide/services/ntp/chronyd_specify_remote_server/ansible/shared.yml:14]: Hello @marcusburghardt
I was also trying to use "plain vanilla" ansible - but I did not find a way to implement this logic - to have a loop and inside the loop to have 2 conditions - 1st which checks for existence of a line which has a time server as a sub-string, 2nd which checks if the line begins with the word "server" and  if both conditions are not met to be a added a line to the configuration file which starts with word server, has a space after, and after the space we have the time server. Would be possible to propose me a working example in ansible which covers this logic? It will help me a lot.
All the best Rumen

**Comment by rumch-se** [comment]
Hello @marcusburghardt

I need advice if a create a PR which checks exactly XCCDF variable. There are some rules which use var_ref in /oval/shared.xml. Because in our case the variable is: time_server1, time_server2, time_server3, time_server4, and my question is the following - May I check with regexp to see if in the configuration file I have one server which as a pattern corresponds to a server in variable, OR it will be better to create 4 variables - for example var_time_server1, var_time_server_2, var_time_server_4 and var_time_server_4 in which we will have time servers. In the second case we will have 4 conditions in the OVAL with "OR" between them and each condition will be dedicated to check if in the config file we have the expected time server.

Have a nice day
Rumen

**Comment by rumch-se** [comment]
Hello @marcusburghardt  and @vojtapolasek 
With my last update I have corrected the ansible part of the rule. Now with this correction
- a missing time server will be added in a correct manner
- wrong rows which include spaces before and after the word server will be corrected
Have  a nice day
Rumen

**Comment by vojtapolasek** [comment]
Hello @rumch-se  and thank you for the improvement. This looks much better. I still need to test if it works correctly. But please introduce this change as completely new rule. This behavior diverts from the original thought behind this already existing rule (and accompanied benchmarks).
Also do not forget about OVAL.

**Comment by rumch-se** [comment]
Hello @vojtapolasek
I think we need to keep bash part. For ansible - you can deside. The new ansible part is shorter than the original and it also works as bash part.  
Have a nice day
Rumen

**Comment by vojtapolasek** [comment]
Hello @rumch-se, unfortunately, I think I don't agree with changes to remediations you propose.
As far as I understand the rule (see my comments above), the rule should not perform remediation if the file /etc/chrony.contains a server or pool directive with any server specified. Your proposed remediations are not doing that - at least from my quick testing, they add servers even in case a pool is defined.
I think this comment is still valid:
https://github.com/ComplianceAsCode/content/pull/10278#discussion_r1126531665

**Comment by rumch-se** [comment]
Hello @vojtapolasek, 
 
We did a lot of exchange about this PR. 

I do believe that you tested only bash remediation part taken from my last commit 

IFS="," read -a SERVERS <<< $var_multiple_time_servers
for srv in "${SERVERS[@]}"
do
   NTP_SRV=$(grep -w $srv $config_file)
   if [[ ! "$NTP_SRV" == "server "* ]]
   then
     time_server="server $srv"
     echo $time_server >> "$config_file"
   fi
done 

According to the logic of bash part - the script is looking into configuration file about the existence of a server described into the variable. If such a line exists it will check if the line begins with the word server. It does not check if the line begins with the word pool. If the line does not begins with the word server , or  we have a space in front of the word server  - this will be a wrong configuration record and a new line will be added. The new line will have the following structure: server, one space, server taken from the variable. These actions will be repeated for each server described into the variable. If your configuration file has some lines which include some spaces - before the word server, or after it - these records will be treated as wrong. If we have word server followed with a server taken from variable without space - this is also a wrong record.

Please show me your initial configuration file and in which cases the bash script adds the records incorrectly. I need to test by myself.

Have a nice day
Rumen

**Comment by vojtapolasek** [comment]
Hello @rumch-se , I did the following test. Create a file /etc/chrony.conf with this single line:
```
pool 2.blabla.pool.ntp.org iburst
```

1. Build the product from your branch and perform the scan of chronyd_specify_remote_server rule.
2. The rule is reported as passing - and this is in the spirit of the rule chronyd_specify_remote_server which allows ANY line which starts with server or pool keyword.
3. Go to build/bash and select a playbook for some profile which contains the rule. Copy out the remediation part for this rule. This step is done because we need the full remediation which would be executed by the scanner if the rule would be failing.
4. save the remediation into a file.
5. run the file - this simulates the case when the rule would be failing and the scanner would be run with --remediate option.
6. cat /etc/chrony.conf
7. You see that the original line stayed there and it is being followed by more lines which contain servers defined in the variable.

I think this is not expected. The rule is written in a way that it passes in case there is a line in /etc/chrony.conf which starts with server followed by one or more spaces and then by anything. And both BAsh and Ansible should be aligned with the rest of the rule, e.g. description and OVAL. I understand that that is not what you expect from such kind of rule, that's why you created the https://github.com/ComplianceAsCode/content/pull/10353.
Please align remediations with the rest of the rule so that they do not modify the file in the case that the rule is not failing.

**Comment by rumch-se** [comment]
Hello @vojtapolasek and @marcusburghardt  

Thank you for your feedback. 

1/At the moment the bash remediation, proposed by me, is looking only for word server, not for word pool. Because of that it will add server(s) to the configuration file. If you wish you can replace in your tests the word pool with the word server and after to test how the solution will work. In my PR 10353 I am looking for both words server and pool.

2/If the rule allows "ANY line which starts with server or pool keyword" we can have the following case

If in the configuration includes:

server  2.blabla.server.ntp.org siburst
pool  1.blabla.pool.ntp.org piburst

the OVAL check will pass, because it will show that the configuration file is correct 

BUT

if the variable defines for a vendor (redhat) these records:  
for server:  2.blabla.redhat_server.ntp.org siburst
for pool:    1.blabla.redhat_pool.ntp.org siburst

these configurations will be not applied, because the rule passed.

In other words as a result we will have on host with redhat OS configuration file which is not correct. 

If for some reason the configuration is modified by a attacker by entering server names controlled by him  the rule will not find this. The attackers "have interest" to modify the time sources. This will open the door for spoofing.

3/Why we define servers via the variable, and we don't check for them via OVAL?    

Have a nice day

Rumen

**Comment by vojtapolasek** [comment]
Hello @rumch-se  and sorry for the delay, I was out for a week.
Honestly, I am a bit lost when trying to find out what you are trying to accomplish in this PR. The original idea was to fix the Bash remediation which incorrectly inserted time servers multiple times. That was a wrong behavior and I believe it should be fixed.
I also understand your concern about security impact of THIS rule, it is valid. However, I think that the way to ensure that servers are configured correctly is NOT to modify behavior of this rule, but to create a new rule and if needed, replace the rule modified in this PR by the new rule. You already did that: https://github.com/ComplianceAsCode/content/pull/10353
And I am happy to review and eventually merge the new rule, because I definitely see a value in it.
However, I still stand behind this comment: https://github.com/ComplianceAsCode/content/pull/10278#discussion_r1126531665
Changing how the rule is remediated or checked just changes the whole logic of the rule, which I believe is not desired from user experience point of view.

**Comment by rumch-se** [comment]
Hello @vojtapolasek,

After so long conversation, I will prefer to use for SUSE products the rule covered by   https://github.com/ComplianceAsCode/content/pull/10353
According to this PR - I think it will better to keep the rule as it is. 

Have a nice day
Rumen

**Comment by vojtapolasek** [comment]
@rumch-se  okay. So what are your plans with this PR? Do you plan to include the change I suggest in the comment?

**Comment by marcusburghardt** [comment]
@rumch-se , I propose you to close this PR in favor of https://github.com/ComplianceAsCode/content/pull/10591

**Comment by rumch-se** [comment]
Hello 
I close this PR in favor of https://github.com/ComplianceAsCode/content/pull/10591
Have a nice day
Rumen
